docs/transformations.md

```diff
@@ -0,0 +1,91 @@
+# Transformations {#transformations}
+<!--
+Copyright (C) The DDC development team, see COPYRIGHT.md file
+
+SPDX-License-Identifier: MIT
+-->
+
+On the top of DDC core - which provides the basic tools for the needs of performance-portable physical simulations
+(describing meshes with *DiscreteDomain* and representing physical fields with *Chunk*) - are build computation kernels
+to perform transformations. At the moment, the two available transformations are Discrete Fourier Transform and Spline methods.
+
+Those methods are performance portable (supported with CPU Serial, CPU OpenMP, CUDA and HIP backends). They are used
+in *examples/heat_equation_spectral.cpp* (for DFT) and *examples/characteristics_advection.cpp* (for splines).
+A particular use-case of splines is benchmarked in *benchmarks/splines.cpp*.
+
+## General presentation of transformations
+
+### Computation kernels dedicated to transformations in DDC
+
+A transformation is a change of basis, going from a function space to another. Because the available memory is finite,
+the basis are necessarily finite (which means it contains a finite set of basis functions, but those basis functions
+may be continuous).
+
+The use-cases of transformations for simulation or signal processing needs are plethoric: interpolation, spectral methods,
+finite element methods, filtering (compression), signal analysis (post-process)...
+
+Every basis have their own specificities (ie. orthogonality, n-derivability, etc...) but as they belong to the same frame
+of set of functions which form a basis, they have in common a formalism and a terminology. However, this is currently
```

Comment:
```suggestion
or set of functions which form a basis, they have in common a formalism and a terminology. However, this is currently
```

Reply:
but as they belong to the same frame of "set of functions which form a basis"

En français on a "base de fonctions" mais en anglais je trouve pas d'équivalent plus compact

Reply:
Ah ok. In that case I think you can remove "frame of" it doesn't mean anything here.

"base de fonctions" = function space basis ?

Reply:
Probably yes, I find the "space" here a bit confusing because in my understanding ie. Function space of Fourier transform (abstract high-dimensional space generated from all e^ikx) is not the same as Fourier space (same dimensionality as real space whose points are identified with wavevector). That's why i'm trying to avoid using the word "space". What do you think ?

Reply:
Function space is a term from linear algebra (https://en.wikipedia.org/wiki/Vector_space#Function_spaces). It is a kind of vector space.

> A vector space (also called a linear space) is a set whose elements, often called vectors, may be added together and multiplied ("scaled") by numbers called scalars.

Usually when we think of vector spaces we think of a finite-dimensional vector space. I.e. a vector space with a finite number of basis elements. However it is also possible to have an infinite-dimensional vector space.

The function space of Fourier transform is a finite-dimensional function space whose basis elements are
![image](https://github.com/CExA-project/ddc/assets/25363473/9e67fb88-8377-449b-a889-54562de8f943)

while the Fourier space is an infinite-dimensional function space with an infinite number of basis elements
![image](https://github.com/CExA-project/ddc/assets/25363473/1823f109-b008-4316-bc6d-0281317d5006) $n \in [-\infty, ..., \infty]$

Reply:
A Discrete Fourier transform takes you from the space of real functions to a subset of the Fourier space so I don't think this is necessarily misleading for those who are familiar with the theory of Fourier transforms and DFT.

Also beware, in IT most of the time when we talk about "Fourier transforms" what we mean is "Discrete Fourier Transform". Mathematically speaking, a Fourier transform is calculated via an integral and the result is indeed something in Fourier space. However when we calculate this on a computer we necessarily have a "Discrete Fourier Transform" and therefore map to a subset of this space

Reply:
>  ie. Function space of Fourier transform (abstract high-dimensional space generated from all e^ikx) is not the same as Fourier space (same dimensionality as real space whose points are identified with wavevector)

If by "high-dimensional space" you mean "infinitely high-dimensional space" then I don't think I agree with this statement

Reply:
Otherwise maybe this is clearer:

> Every basis has its own specificities (ie. orthogonality, n-derivability, etc...) but as they are all basis functions,
> they have in common a formalism and a terminology. However, this is currently

Reply:
By "high-dimensional space" I mean space with as much dimensions as their are basis functions (N in your formula). Infinite-dimensional only in the continuous case then.

I think I do not agree with this:
"
the Fourier space is an infinite-dimensional function space with an infinite number of basis elements
image 
"

To me the Fourier space is just the space in which $\tilde{f}(\vec(k))$ lies:

"Fourier space (or frequency domain or Fourier domain) is an abstract "space" into which the [Fourier transform] maps a function, consisting of the [amplitude] and phase of the sine function at various [frequencies] that sum to produce the same shape" http://astro.vaporia.com/start/fourierspace.html

So Fourier space maps the Function space of Fourier transform, but they are not the same.

Reply:
I see what you mean `k` is not really a function it is just an object in Fourier space. So Fourier space is a Vector space not a Function space.

The basis doesn't belong to Fourier space it is in a Function space. I agree that these concepts are hard to keep straight but I'm not sure it is the very generic word "space" that causes these problems. In maths "space" just means that you can do `a*b+c` for elements `b` and `c` which belong to the space. This is true for Function space bases and for elements of Fourier space.

What term are you using for the basis elements  ![image](https://github.com/CExA-project/ddc/assets/25363473/122d0391-5ee9-4e30-888a-597d2b1d3da7) ?
I think that it is this that is hard to define as it is unrelated to the Fourier transform

Reply:
I've just realised that you mean "framework"

Reply:
bases plural